src/lib/sw_engine/tvgSwFill.cpp

```diff
@@ -123,7 +114,21 @@ bool _prepareLinear(SwFill* fill, const LinearGradient* linear, const Matrix* tr
 
     fill->linear.dx /= fill->linear.len;
     fill->linear.dy /= fill->linear.len;
-    fill->linear.offset = -fill->linear.dx * x1 -fill->linear.dy * y1;
+    fill->linear.offset = -fill->linear.dx * x1 - fill->linear.dy * y1;
+
+    if (transform) {
+        Matrix invTransform;
+        if (!inverse(transform, &invTransform)) return false;
```

Comment:
if here it needs the inversed matrix, what about getting it at fillGenColorTable() ? 
propbably radialGradient might also need it?